ch04/base.md

```diff
@@ -37,75 +37,85 @@ resulting tree.
 
 The implicit ``html` and `body` (and `head` when needed) tags are added:
 
-	>>> test_parse("<html><body>test</body></html>")
-	 <html>
-	   <body>
-	     'test'
+        >>> test_parse("<html><body>test</body></html>")
+         <html>
+           <body>
+             'test'
 
 Quotation marks around `Text` objects should be different depending on if
 there are quotes within:
 
-	>>> test_parse("<html><body>baker's dozen</body></html>")
-	 <html>
-	   <body>
-	     "baker's dozen"
+        >>> test_parse("<html><body>baker's dozen</body></html>")
+         <html>
+           <body>
+             "baker's dozen"
 
 Missing tags are added in:
 
-	>>> test_parse("test")
-	 <html>
-	   <body>
-	     'test'
+        >>> test_parse("test")
+         <html>
+           <body>
+             'test'
 
-	>>> test_parse("<body>test")
-	 <html>
-	   <body>
-	     'test'
+        >>> test_parse("<body>test")
+         <html>
+           <body>
+             'test'
 
 Head tags are put in the head, and other tags, such as `div`, are put
 in the body. Also, tags such as `base` are self-closing:
 
-	>>> test_parse("<base><basefont></basefont><title></title><div></div>")
-	 <html>
-	   <head>
-	     <base>
-	     <basefont>
-	     <title>
- 	   <body>
- 	     <div>
+        >>> test_parse("<base><basefont></basefont><title></title><div></div>")
+         <html>
+           <head>
+             <base>
+             <basefont>
+             <title>
+           <body>
+             <div>
 
 Missing end tags are added:
 
-	>>> test_parse("<div>text")
-	 <html>
-	   <body>
-	     <div>
-	       'text'
+        >>> test_parse("<div>text")
+         <html>
+           <body>
+             <div>
+               'text'
 
 Attributes can be set on tags:
 
-	>>> test_parse("<div name1=value1 name2=value2>text</div")
-	 <html>
-	   <body>
-	     <div name1="value1" name2="value2">
-	       'text'
+        >>> test_parse("<div name1=value1 name2=value2>text</div")
+         <html>
+           <body>
+             <div name1="value1" name2="value2">
+               'text'
+
+Test that the italic `</i>` tag behaves correctly:
```

Comment:
This isn't what we should test for: we should be testing that `</i>` in fact makes text stop being italic. So instead of `test_parse` we should parse it, make a `Layout` out of it, and print the `display_list`, which should show two words with the first one being italic but the second one not being italic. You can put it in the "Testing Layout" sections and model it off the first test in that section.